pkg/scheduler/scheduler_test.go

```diff
@@ -953,27 +953,19 @@ func TestSchedulerWithVolumeBinding(t *testing.T) {
 				t.Fatalf("scheduling timeout after %v", wait.ForeverTestTimeout)
 			}
 			stopFunc()
-			// Wait for scheduling to return an error
+			// Wait for scheduling to return an error or succeed binding.
 			select {
 			case err := <-errChan:
 				if item.expectError == nil || !reflect.DeepEqual(item.expectError.Error(), err.Error()) {
 					t.Errorf("err \nWANT=%+v,\nGOT=%+v", item.expectError, err)
 				}
-			case <-time.After(chanTimeout):
-				if item.expectError != nil {
-					t.Errorf("did not receive error after %v", chanTimeout)
-				}
```

Comment:
move this to the `<-time.After(chanTimeout)` case

Reply:
Better done after the select.